lib/net/ssh/proxy/command.rb

```diff
@@ -56,8 +56,8 @@ def open(host, port, connection_options = nil)
       }
       begin
         io = IO.popen(command_line, "r+")
-        if result = Net::SSH::Compat.io_select([io], nil, [io], 60)
-          if result.last.any?
+        if result = self.wait_readable(60)
+          if result == false
```

Comment:
```ruby
if result = self.wait_readable(60)
  if result == false
     # cannot be reached
  end
end
```
looks fishy to me.
I'm not sure on which cases can IO.select return an exceptional event for pipes but those are not handled by wait_readable for sure.